File: 118. Pascal's Triangle/118.go
// 118. Pascal's Triangle solution in GoLang
// https://leetcode.com/problems/pascals-triangle/description/

package leetcode

func generate(numRows int) [][]int {
    pascal := make([][]int, numRows)

    for i := 0; i <  numRows; i++ {
        pascal[i] = make([]int, i+1)
        pascal[i][0] = 1

        for j := 1; j < i; j++ {
            pascal[i][j] = pascal[i-1][j-1] + pascal[i-1][j]
        }

        pascal[i][i] = 1
    }

    return pascal
}

File: 118. Pascal's Triangle/118.java
// 118. Pascal's Triangle solution in Java
// https://leetcode.com/problems/pascals-triangle/description/

class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> pascal = new ArrayList<List<Integer>>();
        List<Integer> prev_row = new ArrayList<>();
        for (int i = 0; i < numRows; i++) {
            List<Integer> row = new ArrayList<>();
            row.add(1);

            for (int j = 1; j < i; j++) {
                row.add(prev_row.get(j-1) + prev_row.get(j));
            }
            
            if (i != 0) {
                row.add(1);
            }
            pascal.add(row);
            prev_row = row;
        }

        return pascal;
    }
}